### PR TITLE
Add noexcept for RCP

### PR DESCRIPTION
**Pull request by isuruf**
This has a small performance improvement as std containers will only use moves if the move constructor is `noexcept`

### CODE REVIEW

**Comment by isuruf** [comment]
@certik, can you review?

File: symengine/symengine_rcp.h [hunk 1]
```diff
@@ -86,14 +86,14 @@ class RCP {
...
         rp.ptr_ = nullptr;
     }
     // Move constructor
-    template<class T2> RCP(RCP<T2>&& r_ptr) : ptr_(r_ptr.get()) {
+    template<class T2> RCP(RCP<T2>&& r_ptr) SYMENGINE_NOEXCEPT : ptr_(r_ptr.get()) {
         r_ptr._set_null();
     }
-    ~RCP() {
+    ~RCP() SYMENGINE_NOEXCEPT {
         if (ptr_ != nullptr and --(ptr_->refcount_) == 0) delete ptr_;
```
Comment (certik) [inline symengine/symengine_rcp.h]: Isn't a destructor automatically noexcept? Also, deleting a class can actually raise an exception, but in this case I think it aborts the code, which is what we want anyway.

**Comment by certik** [comment]
It looks good to me.

File: symengine/symengine_rcp.h (hunk 1 above)
Comment (isuruf) [inline symengine/symengine_rcp.h]: Yes, a destructor is automatically `noexcept`, but some `gcc` versions had a bug that this was needed.

Comment (certik) [inline symengine/symengine_rcp.h]: Ok, that's fine then.

**Pull request merged**